nautobot/extras/management/__init__.py

```diff
@@ -77,43 +81,78 @@
     "Maintenance": "Unit is under maintenance",
     "Offline": "Unit is offline",
     "Planned": "Unit has been planned",
+    "Primary": "Unit is primary",
     "Provisioning": "Circuit is being provisioned",
     "Reserved": "Unit is reserved",
     "Retired": "Location has been retired",
+    "Secondary": "Unit is secondary",
     "SLAAC": "Dynamically assigned IPv6 address",
     "Staged": "Unit has been staged",
     "Staging": "Location is in the process of being staged",
 }
 
+# List of 2-tuples of (model_path, choiceset)
+# Add new mappings here as other models are supported.
+ROLE_CHOICESET_MAP = {
+    "extras.ContactAssociation": extras_choices.ContactAssociationRoleChoices,
+}
+
+# Map of role name -> default hex_color used when importing color choices in `export_roles_from_choiceset()`.
+# Only a small subset of colors are used by default as these originally were derived from Bootstrap CSS classes.
+ROLE_COLOR_MAP = {
+    "Administrative": ColorChoices.COLOR_BLUE,
+    "Billing": ColorChoices.COLOR_GREEN,
+    "Support": ColorChoices.COLOR_YELLOW,
+    "On Site": ColorChoices.COLOR_BLACK,
+}
+
+# Map of role name -> description used when importing role choices in `export_roles_from_choiceset()`.
+ROLE_DESCRIPTION_MAP = {
+    "Administrative": "Unit plays an administrative role",
+    "Billing": "Unit plays a billing role",
+    "Support": "Unit plays a support role",
+    "On Site": "Unit plays an on site role",
+}
+
 
-#
-# Statuses
-#
+### Migration helper methods to populate/clear statuses and roles
 
 
-def populate_status_choices(apps=global_apps, schema_editor=None, **kwargs):
+def populate_metadata_choices(apps=global_apps, schema_editor=None, **kwargs):
```

Comment:
IMO we should have a `populate_status_choices` and `populate_role_choices`. The extra few lines of code is worth it to make these easier to understand and use.

Reply:
We should also update the docstring to describe the defined kwargs and document that the `**kwargs` will be passed through to `_create_custom_role_or_status_instances`